### ISSUE TITLE
AppInsights ApiKey keeps getting `create-replacement` issue after first create

### ISSUE THREAD

**Issue by lyeef** [pulumi/pulumi-azure#1008]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

I'm trying to create an apikey for an appinsights following this [guide](https://www.pulumi.com/registry/packages/azure/api-docs/appinsights/apikey/). The first create looks good, but all the subsequent pulumi preview/up, I am getting issue where it will be `create-replacement` but actually nothing has changed in the appinsights nor the appinsights apikey settings. This is the pulumi preview diff logs:
```
++ azure:appinsights/apiKey:ApiKey (create-replacement)
    [id=/subscriptions/[id]/resourcegroups/[resourceGroupName]/providers/microsoft.insights/components/[resourceGroupName]/apiKeys/[keyId]]
    __meta               : "{"....":{"create":1800000000000,"delete":1800000000000,"read":300000000000,"update":1800000000000},"schema_version":"1"}"
    apiKey               : "[secret]"
    applicationInsightsId: "/subscriptions/[id]/resourceGroups/[resourceGroupName]/providers/Microsoft.Insights/components/[resourceGroupName]" => "/subscriptions/[id]/resourceGroups/[resourceGroupName]/providers/microsoft.insights/components/[resourceGroupName]"
    id                   : "/subscriptions/[id]/resourcegroups/[resourceGroupName]/providers/microsoft.insights/components/[resourceGroupName]/apiKeys/[keyId]"
    writePermissions     : []
```

### Steps to reproduce

These are my initial codes:
```
const appInsights = new azure.appinsights.Insights(
    componentName,
    {
        resourceGroupName: resourceGroupName,
        location: location,
        name: componentName,
        applicationType: "web"
    },
    { parent: resourceGroup });

const appInsightsApiKey = new azure.appinsights.ApiKey(
    componentName, {
        name: componentName,
        applicationInsightsId: appInsights.id,
        readPermissions: ["aggregate", "api", "draft", "extendqueries", "search"],
    },
    { parent: appInsights });
```

I was able to "hackily" solved the issue by replacing the `applicationInsightsId` case difference with code:
```
const appInsightsApiKey = appInsights.id.apply(id => new azure.appinsights.ApiKey(
    componentName, {
        name: componentName,
        applicationInsightsId: id.replace(/microsoft.insights/g, "Microsoft.Insights"),
        readPermissions: ["aggregate", "api", "draft", "extendqueries", "search"],
    },
    { parent: appInsights }));
```

I just made the changes (replacing the the case) based on the pulumi preview diff logs but have no idea why it had to be done this way. It would be great if someone can explain whats happening here. thanks!

Expected: appInsights apikey should remain unchanged on second and subsequents runs
Actual: appInsights apikey keeps getting `create-replacement` issue on second and subsequents runs

**Comment by leezen** [comment]
I think the workaround you have is going to be good for now as it looks like this is a bug upstream (https://github.com/hashicorp/terraform-provider-azurerm/issues/13348). I suspect it's possibly related to reading the resource back from the provider before writing it to state and having had that property canonicalized (vs. what was provided as input).

**Comment by scp-mb** [comment]
Guessing this is the same issue that's causing refreshes to also fail on one of our stacks

```
azure:appinsights:ApiKey (xxxxxxxxxxxx):
    error: Preview failed: refreshing urn:pulumi:xxxxxxxxx::xxxxxxxxxxxx::azure:appinsights/apiKey:ApiKey::xxxxxxxxxxxxxxxxxx: 1 error occurred:
        * ID was missing the `apiKeys` element
```

**Comment by mikhailshilkov** [comment]
I believe this has been fixed in [v5.27.0](https://github.com/pulumi/pulumi-azure/releases/tag/v5.27.0) (fixed upstream in https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.33.0)
